scripts/export_sqlite_to_workspace.py

```diff
@@ -0,0 +1,259 @@
+
+    ### settings
+    rows = query_db('select * from settings')
+    # make current into a dic
+    csett = {}
+    for row in rows:
+        csett[row['key'].lower()] = row['value']
+
+    # defaults
+    defSett = dict(
+        modbus_enabled = False,
+        modbus_port = 502,
+
+        dnp3_enabled = False,
+        dnp3_port = 20000,
+
+        enip_enabled = False,
+        enip_port = 5555,
+
+        pstorage_enabled = False,
+        pstorage_polling = 10,
+
+        slave_polling = 100,
+        slave_timeout = 1000,
+
+        start_run_mode = False,
+    )
+
+    # loop defaults and check current in case some absent
+    # note: introducting _active flags
+    sett = {}
+    for k, dv in defSett.items():
+
+        parts = k.split("_")
+        cv = csett[k] # current value
+        #print(k, v, cv, parts)
+
+        if parts[1] == "port":
+            if csett.get(k) == "disabled":
+                sett[parts[0] + "_enabled"] = False
+                sett[k] = dv # copy default port no
+            else:
+                sett[parts[0] + "_enabled"] = True
+                sett[k] = int(csett[k])
+        elif k == "pstorage_polling":
+            if csett.get(k) == "disabled":
+                sett[parts[0] + "_enabled"] = False
+                sett[k] = dv  # copy default port no
+            else:
+                sett["pstorage_enabled"] = True
+                sett[k] = int(cv)
+        elif k == "start_run_mode":
+            sett["start_run_mode"] = bool(cv)
+        elif parts[1] == "enabled":
+            sett[k] = bool(cv)
+        else:
+            sett[k] = int(cv)
+
+    write_json(None, "settings.json", sett)
+    write_json(None, "settings.default.json", defSett)
+
+    etc_dir = os.path.abspath(os.path.join(openplc, "etc"))
+    for fn in "active_program", "openplc_driver","openplc_platform":
+        fp = os.path.join(etc_dir, fn)
+        contents = ""
+        with open(fp, "r") as f:
+            contents = f.read()
+            f.close()
+        sf = os.path.join(workspace, "%s.txt" % fn)
+        with open(sf, "w") as f:
+            f.write(contents.strip())
+            f.close()
+            print("Wrote: %s" % sf)
+
+    ## uploads demo
+    up = os.path.join(workspace, "uploads")
+    if not os.path.exists(up):
+        os.mkdir(up)
+
+    st = os.path.join(openplc, "etc", "st_files", "blank_program.st")
+    tar = os.path.join(up, "%s.blank_program.st" % gen_uid())
+    shutil.copyfile(st, tar)
+    print("Copied: %s" % tar)
+
+    ## archive
+    conn.close()
+
+    ark = os.path.join(workspace, "archive")
+    if not os.path.exists(ark):
+        os.mkdir(ark)
+    fn = "%s.%s" % (gen_uid(), os.path.basename(database))
+    dbak = os.path.join(ark, fn)
+    shutil.copyfile(database, dbak)
+    print("Copied: %s" % dbak)
+
+
+
+    print("DONE :-)")
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Export sqlite to workspace folder")
```

Comment:
This code should be in a function. Normally I would call it main().

Why?

Because all of the variables here, for example "parser" and "args" get declared with module scope, when what you really wanted was function scope.